pkg/operator/ceph/object/user/controller.go

```diff
@@ -152,6 +173,21 @@ func (c *ObjectStoreUserController) createUser(context *clusterd.Context, u *cep
 	}
 	objContext := object.NewContext(context, u.Spec.Store, u.Namespace)
 
+	// Wait with exponential backoff for the objectstore to be created
+	err := wait.ExponentialBackoff(backoff, func() (ok bool, err error) {
+		_, err = context.RookClientset.CephV1().CephObjectStores(u.Namespace).Get(u.Spec.Store, metav1.GetOptions{})
```

Comment:
This will only wait for the object store CR to exist. In addition we will need to check that rook has initialized the object store. When rook is initializing the object store the last thing it will do is [start the rgw pod(s)](https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/rgw.go#L95), so you can check that the rgw pod is in the Running state. You could query for pods matching the label `rgw=<store name>` and `app=rook-ceph-rgw`.

Reply:
can we make use of https://github.com/rook/rook/blob/master/pkg/operator/ceph/object/rgw.go#L303 instead?
To check for rgw pods, I'd have to know the name which might change. Solution would be to Query for deployments. This is what the above mentioned function does.

Reply:
That checks if it is configured, but not if it is running. If you check for a running rgw pod then it will give you more accurate info about the service being available. You don't actually need to know the name of the pod. If you call `List()` on pods you can give it the `ListOptions`, which allows you to match based on the labels.

Reply:
ack.